cmd/channel/email/main.go

```diff
@@ -40,12 +46,46 @@ func (ch *Email) SendNotification(req *plugin.NotificationRequest) error {
 
 	plugin.FormatMessage(&msg, req)
 
-	err := smtp.SendMail(ch.GetServer(), nil, ch.From, to, bytes.ReplaceAll(msg.Bytes(), []byte("\n"), []byte("\r\n")))
+	c, err := smtp.Dial(ch.GetServer())
 	if err != nil {
 		return err
 	}
+	defer c.Close()
 
-	return nil
+	if ch.UseTLS {
+		if err := c.StartTLS(&tls.Config{ServerName: ch.Host, MinVersion: tls.VersionTLS12}); err != nil {
+			return err
+		}
+	}
+
+	if ch.Password != "" {
+		if err := c.Auth(smtp.PlainAuth("", ch.From, ch.Password, ch.Host)); err != nil {
+			return err
+		}
+	}
+
+	if err = c.Mail(ch.From); err != nil {
+		return err
+	}
+	for _, addr := range to {
+		if err = c.Rcpt(addr); err != nil {
+			return err
+		}
+	}
+	w, err := c.Data()
+	if err != nil {
+		return err
+	}
+
+	if _, err = w.Write(bytes.ReplaceAll(msg.Bytes(), []byte("\n"), []byte("\r\n"))); err != nil {
```

Comment:
Why replace the newline characters? Maybe it might be better to define a proper Content-Type.

Btw, the handmade email header field generation looks a bit thin and not RFC-compliant, especially regards special characters and the like.

https://github.com/Icinga/icinga-notifications/blob/64067793a9fccc42498051561eb91d78ea55638c/cmd/channel/email/main.go#L44-L45